File: app/src/main/java/group0/eduworld/BookingFragment.java
package group0.eduworld;

import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

public class BookingFragment extends Fragment {
    private int bookingState = 0;

    @Override
    public void onStart() {
        super.onStart();

        View view = getView();
        assert view != null;

        view.findViewById(R.id.choiceGroup).setVisibility(View.GONE);
        view.findViewById(R.id.editButton).setVisibility(View.GONE);
        view.findViewById(R.id.cancelledLabel).setVisibility(View.GONE);
        switch(bookingState){
            case 0:
                getView().findViewById(R.id.choiceGroup).setVisibility(View.VISIBLE);
                break;
            case 1:
                getView().findViewById(R.id.editButton).setVisibility(View.VISIBLE);
                break;
            case 2:
                getView().findViewById(R.id.cancelledLabel).setVisibility(View.VISIBLE);
                break;
        }
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        return inflater.inflate(R.layout.fragment_home, container, false);
    }
}

File: app/src/main/java/group0/eduworld/BookingFragment.kt
package group0.eduworld

import android.content.Context
import android.net.Uri
import android.os.Bundle
import android.support.v4.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.util.AttributeSet
import kotlinx.android.synthetic.main.fragment_booking.*


// TODO: Rename parameter arguments, choose names that match
// the fragment initialization parameters, e.g. ARG_ITEM_NUMBER
private const val ARG_PARAM1 = "param1"
private const val ARG_PARAM2 = "param2"

/**
 * A simple [Fragment] subclass.
 * Activities that contain this fragment must implement the
 * [BookingFragment.OnFragmentInteractionListener] interface
 * to handle interaction events.
 * Use the [BookingFragment.newInstance] factory method to
 * create an instance of this fragment.
 *
 */
class BookingFragment : Fragment() {
    private var bookingState: Int = 0

    // TODO: Rename and change types of parameters
    private var param1: String? = null
    private var param2: String? = null
    private var listener: OnFragmentInteractionListener? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        arguments?.let {
            param1 = it.getString(ARG_PARAM1)
            param2 = it.getString(ARG_PARAM2)
        }
    }

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        // Inflate the layout for this fragment
        return inflater.inflate(R.layout.fragment_booking, container, false)
    }

    /**
     * Parse attributes during inflation from a view hierarchy into the
     * arguments we handle.
     */
    override fun onInflate(context: Context, attrs: AttributeSet, savedInstanceState: Bundle?) {
        super.onInflate(context, attrs, savedInstanceState)

        val a = context.obtainStyledAttributes(attrs, R.styleable.BookingFragment)

        bookingState = a.getInt(R.styleable.BookingFragment_state, 0)

        choiceGroup?.visibility = if (bookingState == 0) View.VISIBLE else View.GONE
        editButton?.visibility = if (bookingState == 1) View.VISIBLE else View.GONE
        cancelledLabel?.visibility = if (bookingState == 2) View.VISIBLE else View.GONE

        a.recycle()
    }


    // TODO: Rename method, update argument and hook method into UI event
    fun onButtonPressed(uri: Uri) {
        listener?.onFragmentInteraction(uri)
    }

    override fun onAttach(context: Context) {
        super.onAttach(context)
        if (context is OnFragmentInteractionListener) {
            listener = context
        } else {
            throw RuntimeException(context.toString() + " must implement OnFragmentInteractionListener")
        }
    }

    override fun onDetach() {
        super.onDetach()
        listener = null
    }

    /**
     * This interface must be implemented by activities that contain this
     * fragment to allow an interaction in this fragment to be communicated
     * to the activity and potentially other fragments contained in that
     * activity.
     *
     *
     * See the Android Training lesson [Communicating with Other Fragments]
     * (http://developer.android.com/training/basics/fragments/communicating.html)
     * for more information.
     */
    interface OnFragmentInteractionListener {
        // TODO: Update argument type and name
        fun onFragmentInteraction(uri: Uri)
    }

    companion object {
        /**
         * Use this factory method to create a new instance of
         * this fragment using the provided parameters.
         *
         * @param param1 Parameter 1.
         * @param param2 Parameter 2.
         * @return A new instance of fragment BookingFragment.
         */
        // TODO: Rename and change types and number of parameters
        @JvmStatic
        fun newInstance(param1: String, param2: String) =
            BookingFragment().apply {
                arguments = Bundle().apply {
                    putString(ARG_PARAM1, param1)
                    putString(ARG_PARAM2, param2)
                }
            }
    }
}
